src/utilities/messages.ts

```diff
@@ -0,0 +1,8 @@
+import messages from '../config/constants/messages.json';
+
+export const t = (key: string): string => {
```

Comment:
`t` isn't a great function name. Consider renaming this to `errorMessageForError`.

You could also make a helper:
```
export const throwError = (error: ErrorType): void => {
  throw Error(errorMessageForError(error));
}
```

Since you seem to repeat the pattern `throw Error(t('___error_type___`));` a whole lot.

`throwError` could be renamed to `assertError` or `raiseError` to make it more generic.

Also then we can plugin error logging behavior in this helper function (such as printing error messages to console, stack traces, having a unified error type)